connect/runtime/src/main/java/org/apache/kafka/connect/util/TopicAdmin.java

```diff
@@ -0,0 +1,368 @@
+     * This operation requires the broker to support clients performing several administrative operations. If the broker
+     * does not support this capability, this method returns null.
+     *
+     * @param topic the specifications of the topic
+     * @return the topic description of the newly-created or existing topic; null only
+     * if the broker is too old to perform this operation
+     * @throws ConnectException if an error occurs, the operation takes too long, or the thread is interrupted while
+     *                          attempting to perform this operation
+     */
+    public TopicDescription createTopicIfMissing(NewTopic topic) {
+        Map<String, TopicDescription> result = createTopicsIfMissing(topic);
+        return result != null ? result.get(topic.name()) : null;
+    }
+
+    /**
+     * Given the topic definitions, check whether the topics exists and for all those that do not attempt to create the
+     * missing topics. The brokers will use its default topic options for anything not specified in the supplied definitions.
+     * This method returns the topic details for all of the topic included in the supplied descriptions.
+     * <p>
+     * This operation requires the broker to support clients performing several administrative operations. If the broker
+     * does not support this capability, this method returns null.
+     *
+     * @param topics the specifications of the topics
+     * @return the topic description of the newly-created or existing topics, keyed by the topic names; null only
+     * if the broker is too old to perform this operation
+     * @throws ConnectException if an error occurs, the operation takes too long, or the thread is interrupted while
+     *                          attempting to perform this operation
+     */
+    public Map<String, TopicDescription> createTopicsIfMissing(NewTopic... topics) {
+        if (topics == null || topics.length == 0) return Collections.emptyMap();
+
+        Map<String, NewTopic> topicsByName = new HashMap<>();
+        for (NewTopic topic : topics) {
+            if (topic != null) topicsByName.put(topic.name(), topic);
+        }
+        Set<String> allRequestedTopicNames = new HashSet<>(topicsByName.keySet());
+        String bootstrapServers = bootstrapServers();
+
+        try {
+            // Getting topic descriptions for non-existant topics may auto-create them, so first see which of
+            // our topics do already exist
+            Set<String> existingTopicNames = existingTopics(topicsByName.keySet());
+            if (existingTopicNames == null) {
+                // The broker is too old to do this work, so return immediately
+                log.debug("Unable to use Kafka admin client to list existing topics using the brokers at {}", bootstrapServers);
+                return null;
+            }
+
+            // Remove the topics that already exist
+            for (String existingTopicName : existingTopicNames) {
```

Comment:
I agree, except that I wanted a debug statement that printed out the topics that existed. I can rearrange this, though.